File: day01/day01.jl
open("input.txt", "r") do f
    global changes = [parse(Int, line) for line in eachline(f)]
end

println("Final frequency = $(sum(changes))")

unique_ = Set{Int}()
freq = 0
for change in Iterators.cycle(changes)
    global freq += change
    if in(freq, unique_)
        println("First frequency reached twice = $freq")
        break
    else
        push!(unique_, freq)
    end
end

File: day01/day01.py
import itertools

if __name__ == '__main__':
    with open('input.txt', 'r') as f:
        changes = [int(line) for line in f]
    print('Final frequency = {}'.format(sum(changes)))
    freq = 0
    unique = set()
    for change in itertools.cycle(changes):
        freq += change
        if freq in unique:
            print('First frequency reached twice = {}'.format(freq))
            break
        else:
            unique.add(freq)
